File: 3.哈希表/2.有效的字母异位词/383.赎金信.cpp
/*
 * @lc app=leetcode.cn id=383 lang=cpp
 *
 * [383] 赎金信
 */

// @lc code=start
class Solution {
    public:
        bool canConstruct(string ransomNote, string magazine) {
            unordered_map<char, int> hash;
            for (auto c : magazine)
            {
                hash[c] ++;
            }
            for (auto c : ransomNote)
            {
                if (hash.find(c) == hash.end()) return 0;
                hash[c] --;
                if (hash[c] < 0) return 0;
            }
            return 1;
        }
    };
// @lc code=end

File: 面经150/5.哈希表/383.赎金信.py
#
# @lc app=leetcode.cn id=383 lang=python3
#
# [383] 赎金信
#
from mytools import *
# @lc code=start
class Solution:
    def canConstruct(self, ransomNote: str, magazine: str) -> bool:
        hash = Counter(magazine)
        for ch in ransomNote:
            hash[ch] -= 1
            if hash[ch] < 0: return False
        return True
# @lc code=end

